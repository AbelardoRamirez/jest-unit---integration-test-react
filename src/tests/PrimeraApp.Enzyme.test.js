import React from 'react';
import '@testing-library/jest-dom'
const { shallow } = require("enzyme")
const { default: PrimeraApp } = require("../base/PrimeraApp")

/**
 * Prueba con un SnapShot
 * Se compila la primera vez, se debe de presionar la tecla U para realizar un snapshot de
 * lo que se renderiza & poder compararla con diferentes parametros
 */

describe('Prueba con Enzyme en <PrimeraApp />', () => {
    test('Debe de mostrar <PrimeraApp/> Correctamente', () => {
        const saludo = 'Hola, soy Gokun';
        const lista = [1, 2, 3, 4];
        const wrapper = shallow(<PrimeraApp saludo={saludo} lista={lista} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, Soy Gokun2';
        const lista = [1, 2, 3, 4];
        const subtitulo = 'Mi Subtitulo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} lista={lista} subtitulo={subtitulo} />);
        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);
        expect(textoParrafo).toBe(subtitulo);
    });


});
