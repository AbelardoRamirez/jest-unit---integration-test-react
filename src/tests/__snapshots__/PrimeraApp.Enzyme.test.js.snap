// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con Enzyme en <PrimeraApp /> Debe de mostrar <PrimeraApp/> Correctamente 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <h3
      className="h3 font-weight-bold text-center"
    >
      Experimento Funcional Test
    </h3>
    <h2
      className="h2 font-weight-bolder text-center"
    >
      Hola, soy Gokun
      !!!
    </h2>
    <ul>
      <li
        key="0"
      >
        1
         
        Incompleted
      </li>
      <li
        key="1"
      >
        2
         
        OK
      </li>
      <li
        key="2"
      >
        3
         
        Incompleted
      </li>
      <li
        key="3"
      >
        4
         
        OK
      </li>
    </ul>
    <p
      className="font-weight-lighter text-monospace"
    >
      <span
        className="text-success"
      >
        Mi Subtitulo
      </span>
    </p>
  </div>
</div>
`;
